### PR TITLE
RT-624-success-message:Configurable success message

### PR DESCRIPTION
**Pull request by yashpmohod**
## Overview
Add material snackbar to show success or error messages.

image

## How it was tested
Tested Using run cardiovascular-patient-app locally.

## Checklist

- [x] The title contains a short meaningful summary
- [x] I have added a link to this PR in the Jira issue
- [x] I have described how this was tested
- [x] I have included screen shots for changes that affect the user interface
- [x] I have updated unit tests
- [x] I have run unit tests locally
- [ ] I have updated documentation (including README)

### CODE REVIEW

**Review by semicolin — COMMENTED** [review]
I have some reservations about using snackbar for success/error messages. I recall them being rather annoying in sapphire because they obscured other things on the screen, or disappeared before I was done reading the message. The positioning in your screenshot also seems far away from the button that triggered it. Generally, I like to keep messages close to where the user is already looking. It might be worth taking a screen recording and getting some feedback from the product owner.

**Comment by yashpmohod** [comment]
> I have some reservations about using snackbar for success/error messages. I recall them being rather annoying in sapphire because they obscured other things on the screen, or disappeared before I was done reading the message. The positioning in your screenshot also seems far away from the button that triggered it. Generally, I like to keep messages close to where the user is already looking. It might be worth taking a screen recording and getting some feedback from the product owner.

Okay, @semicolin  I change the position of snackbar from end to center.
image

File: projects/cardiovascular-patient-app/src/app/report-bp/report-bp.component.spec.ts [hunk 1]
```diff
@@ -86,4 +93,32 @@ describe('ReportBPComponent', () => {
     await systolicInputHarness.setValue('252');
     expect(systolicFormField?.errors?.['max']).toBeTruthy();
   });
+
+  it('should submit prioer blood pressure Form', async () => {
+    fhirService.createResourceData.and.callThrough().and.returnValue({ resource: 'resource' });
+    fhirService.addPatientData.and.callThrough();
+    spyOn(component, 'onSubmit').and.callThrough();
+    component.onSubmit();
+    expect(component.onSubmit).toHaveBeenCalled();
```
Comment (semicolin) [inline projects/cardiovascular-patient-app/src/app/report-bp/report-bp.component.spec.ts:102]: there's no point in checking whether onSubmit() has been called when you just called it yourself on the previous line.

File: projects/cardiovascular-patient-app/src/app/report-bp/report-bp.component.spec.ts [hunk 2]
```diff
@@ -86,4 +93,32 @@ describe('ReportBPComponent', () => {
...
+    spyOn(component, 'onSubmit');
+    component.onSubmit();
+    el.triggerEventHandler('ngSubmit', null);
+    expect(component.onSubmit).toHaveBeenCalled();
+  })
+
+  it('should submit the form when save button is clicked', async () => {
+    let el = fixture.debugElement.query(By.css('.btn-class'));
+    spyOn(component, 'onSubmit');
+    component.onSubmit();
```
Comment (semicolin) [inline projects/cardiovascular-patient-app/src/app/report-bp/report-bp.component.spec.ts:119]: test code should not call its own spy. this does nothing.

File: projects/cardiovascular-patient-app/src/app/report-bp/report-bp.component.spec.ts [hunk 3]
```diff
@@ -86,4 +93,32 @@ describe('ReportBPComponent', () => {
...
+    (el.nativeElement as HTMLButtonElement).click();
+    fixture.detectChanges()
```
Comment (semicolin) [inline projects/cardiovascular-patient-app/src/app/report-bp/report-bp.component.spec.ts:121]: Can we use the test harness?

File: projects/ngx-charts-on-fhir/src/lib/fhir-data/fhir-data.service.ts [hunk 4]
```diff
@@ -94,6 +99,85 @@ export class FhirDataService {
...
+          }
+        ],
+        "text": "Blood Pressure"
+      },
+      "subject": {
+        "reference": `Patient/${this.client?.patient.id}`
+      },
+      "encounter": {
+        "reference": `Encounter/${this.client?.encounter.id}`
+      },
```
Comment (semicolin) [inline projects/ngx-charts-on-fhir/src/lib/fhir-data/fhir-data.service.ts:137]: This has the same encounter issue that you fixed in your other PR. Why is this PR duplicating that code? There will be merge conflicts.

**Pull request closed without merging**